packages/node/vro-language-server/src/server/request/collection/ServerCollection.ts

```diff
@@ -51,139 +55,120 @@ export class ServerCollection {
-            promise = promise
-                .then(() => {
-                    if (event.isCancellationRequested) {
-                        this.currentStatus.message = "Operation cancelled"
-                        this.currentStatus.finished = true
-                    } else if (!this.currentStatus.finished) {
-                        return op.apply(this)
-                    }
-
-                    return undefined
-                })
-                .catch(errorMessage => {
-                    this.setError(errorMessage)
-                })
-        }
-    }
-
-    async verifyVroConnection() {
-        this.currentStatus.message = "Authenticating..."
-        this.logger.info(this.currentStatus.message)
-
-        try {
-            const responseBody = await this.restClient.getPlugins()
-            for (const plugin of responseBody.plugin) {
-                if (plugin.moduleName === "HintPlugin" && plugin.enabled) {
-                    this.logger.debug("Hint plugin is installed in target vRO: ", plugin)
-                    this.currentStatus.data.hintsPluginBuild = plugin.buildNumber
-                    return
+    async triggerCollection(offline: boolean, event: CancellationToken) {
+        this.logger.info("Triggering vRO server data collection...")
+        this.environment.workspaceFolders.forEach(workspaceFolder => {
+            this.workspaceCollection.triggerCollectionAndRefresh(workspaceFolder)
+        })
+
+        this.restClient
+            .getVersion()
+            .then(res => {
+                this.logger.info(`Connected to vRO. Version: ${res.version}`)
+
+                this.currentStatus = new CollectionStatus()
+                let promise = Promise.resolve()
+                const operations = [this.collectModulesAndActions, this.collectObjects, this.done]
+
+                for (const op of operations) {
+                    promise = promise
+                        .then(() => {
+                            if (event.isCancellationRequested) {
+                                this.currentStatus.message = "Operation cancelled"
+                                this.currentStatus.finished = true
+                            } else if (!this.currentStatus.finished) {
+                                return op.apply(this)
+                            }
+
+                            return undefined
+                        })
+                        .catch(errorMessage => {
+                            this.setError(errorMessage)
+                        })
                 }
-            }
-
-            this.setError("The vRO Hint plug-in is not installed")
-            this.currentStatus.data.hintsPluginBuild = 0
-        } catch (error) {
-            this.setError(error.message)
-            this.currentStatus.data.hintsPluginBuild = -1
-        }
+            })
+            .catch(error => {
+                this.logger.info("Could not connect to vRO. Collecting offline hints collection...")
```

Comment:
What does it mean "Collecting offline hints collection..." it's going to use the data that is already cached locally? If yes - maybe rephrase a bit the message?